tools/topology/topology2/include/pipelines/cavs/src-gain-playback.conf

```diff
@@ -0,0 +1,186 @@
+				out_bit_depth		32
+				out_valid_bit_depth	32
+				dma_buffer_size "$[$ibs * 2]"
+			}
+			Object.Base.audio_format.2 {
+				in_bit_depth		32
+				in_valid_bit_depth	24
+				out_bit_depth		32
+				out_valid_bit_depth	32
+				dma_buffer_size "$[$obs * 2]"
+			}
+			Object.Base.audio_format.3 {
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_bit_depth		32
+				out_valid_bit_depth	32
+				dma_buffer_size "$[$ibs * 2]"
+			}
+
+			# 44.1khz input
+			Object.Base.audio_format.4 {
+				in_rate                 44100
+				in_bit_depth            16
+				in_valid_bit_depth      16
+				out_bit_depth           32
+				out_valid_bit_depth     32
+				dma_buffer_size "$[$obs * 2]"
+			}
+
+			Object.Base.audio_format.5 {
+				in_rate                 44100
+				in_bit_depth            32
+				in_valid_bit_depth      24
+				out_bit_depth           32
+				out_valid_bit_depth     32
+				dma_buffer_size "$[$obs * 2]"
+			}
+
+			Object.Base.audio_format.6{
+				in_rate                 44100
+				in_bit_depth            32
+				in_valid_bit_depth      32
+				out_bit_depth           32
+				out_valid_bit_depth     32
+				dma_buffer_size "$[$obs * 2]"
+			}
+
+			node_type $HDA_HOST_OUTPUT_CLASS
+		}
+
+		src."1" {
+			num_audio_formats 5
+
+			Object.Base.audio_format.1 {}
+
+			Object.Base.audio_format.2 {
+				in_bit_depth            32
+				in_valid_bit_depth      32
+				out_bit_depth           32
+				out_valid_bit_depth     32
+			}
+
+			Object.Base.audio_format.3 {
+				in_rate                44100
+				in_bit_depth           16
+				in_valid_bit_depth     16
+				out_bit_depth          16
+				out_valid_bit_depth    16
+				# src needs extra 4 sample size
+				obs "$[($out_channels * $[($[($out_rate / 1000)] + 4)]) * ($out_bit_depth / 8)]"
+			}
+
+			Object.Base.audio_format.4 {
+				in_rate                 44100
+				in_bit_depth            32
+				in_valid_bit_depth      24
+				out_bit_depth           32
+				out_valid_bit_depth     24
+				obs "$[($out_channels * $[($[($out_rate / 1000)] + 4)]) * ($out_bit_depth / 8)]"
+			}
+
+			Object.Base.audio_format.5 {
+				in_rate                 44100
+				in_bit_depth            32
+				in_valid_bit_depth      32
+				out_bit_depth           32
+				out_valid_bit_depth     32
+				obs "$[($out_channels * $[($[($out_rate / 1000)] + 4)]) * ($out_bit_depth / 8)]"
+			}
+		}
+
+		gain."1" {
+			num_audio_formats 2
+			#16-bit 48KHz 2ch
+			Object.Base.audio_format.1 {}
+			# 32-bit 48KHz 2ch
+			Object.Base.audio_format.2 {
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_bit_depth		32
+				out_valid_bit_depth	32
+			}
+		}
+
+		pipeline."1" {
+			priority	0
+			lp_mode		0
+		}
+	}
+
+	Object.Base {
+		route."1" {
+			source	"copier.host..1"
+			sink	"src..1"
+		}
+
+		route."2" {
+			source  "src..1"
+			sink    "gain..1"
+		}
+	}
+
+	direction	"playback"
+	dynamic_pipeline 1
+	time_domain	"timer"
+	channels	2
+	channels_min	2
+	channels_max	2
+	rate		48000
+	rate_min	16000
```

Comment:
Smallest rate should likely be 8 kHz.

Reply:
and it's inconsistent to gave rate min and max, but only describe formats for the 44.1 kHz case. Either we support all intermediate values or we don't.

Reply:
sure